.github/workflows/build-cygwin.yml

```diff
@@ -0,0 +1,65 @@
+name: Build on Cygwin
+
+on:
+  push:
+    branches:
+      - '4.**'
+      - '5.**'
+      - 'trunk'
+  pull_request:
+
+# Restrict the GITHUB_TOKEN
+permissions: {}
+
+env:
+  # List of test directories for the debug-s4096 and linux-O0 jobs. These
+  # directories are selected because of their tendencies to reach corner cases
+  # in the runtime system.
+  PARALLEL_TESTS: parallel callback gc-roots weak-ephe-final
+  # Fully print commands executed by Make
+  # MAKEFLAGS: V=1
+
+# https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#concurrency
+# Concurrent workflows are grouped by the PR or branch that triggered them
+# (github.ref) and the name of the workflow (github.workflow). The
+# 'cancel-in-progress' option then make sure that only one workflow is running
+# at a time. This doesn't prevent new jobs from running, rather it cancels
+# already running jobs before scheduling new jobs.
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}-${{ github.event_name == 'pull_request' || github.sha }}
+  cancel-in-progress: true
+
+jobs:
+# This job will do the initial build of the compiler (on linux), with flambda on.
+# We then upload the compiler tree as a build artifact to enable re-use in
+# subsequent jobs.
+  cygwin:
+    runs-on: windows-latest
+    steps:
+      - name: Configure EOLs on Cygwin
+        run: |
+          # Ensure that .expected files are not modified by check out
+          # as, in Cygwin, the .expected should use LF line endings,
+          # rather than Windows' CRLF
+          git config --global core.autocrlf input
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          submodules: true
+      - name: Install Cygwin
+        uses: cygwin/cygwin-install-action@v4
+        with:
+          packages: make,bash,gcc-core,gcc-fortran
+          install-dir: 'D:\cygwin'
+      - name: Configure the compiler
+        shell: bash
+        run: >-
+          ./configure --enable-ocamltest --disable-dependency-generation
+      - name: Build the compiler
+        shell: bash
+        run: >-
+          make -j4 V=1
```

Comment:
```suggestion
          make -j$(nproc) V=1
```

Reply:
Note that I don’t intend this CI workflow to be merged in.
IIRC, you ended up suggesting using simply `make -j` anyway to get the best performance on CI machines, didn’t you?

Reply:
Oh, right. Yes, you're right.